### PR TITLE
feat(antd): `autoSubmitClose`, `autoResetForm`, `defaultVisible` features to `useDrawerForm`

### LINKED ISSUE #3643: [FEAT]: add new properties to `useDrawerForm` hook
### Is your feature request related to a problem? Please describe.

In `@pankod/refine-antd`package, `useDrawerForm` is similar to `useModalForm` hook. However, `useDrawerForm` does not have `defaultVisible`, `autoSubmitClose`, and `autoResetForm` properties. I think these properties should be added to `useDrawerForm` hook.

### Describe alternatives you've considered

_No response_

### Additional context

Similar properties are available in `useModalForm` hook in `@pankod/refine-react-table` package. You can see the [useModalForm](https://github.com/refinedev/refine/blob/next/packages/react-hook-form/src/useModalForm/index.ts) hook for more information.

<br />

You can see the [contribution guide](https://refine.dev/docs/contributing/) for more information.

### Describe the thing to improve

- [ ] `useDrawerForm` hook should have `defaultVisible` property. It will be used to set the initial visibility of the drawer.
- [ ] `useDrawerForm` hook should have `autoSubmitClose` property. It will be used to close the drawer after the form is submitted.
- [ ] `useDrawerForm` hook should have `autoResetForm` property. It will be used to reset the form after the form is submitted.

You can find more information about these properties in [useModalForm](https://refine.dev/docs/api-reference/antd/hooks/form/useModalForm/#properties) hook documentation.

### PR DESCRIPTION
**Pull request by salihozdemir**
Add new props to `useDrawerForm`.

close #3643

### CODE REVIEW

File: packages/antd/src/hooks/form/useDrawerForm/useDrawerForm.ts [hunk 1]
```diff
@@ -102,15 +109,15 @@ export const useDrawerForm = <
             ? `drawer-${resource?.identifier ?? resource?.name}-${action}`
             : undefined;
 
-    const [open, setOpen] = useState(false);
-
     React.useEffect(() => {
         if (initiallySynced.current === false && syncWithLocationKey) {
             const openStatus = parsed?.params?.[syncWithLocationKey]?.open;
```
Comment (BatuhanW) [inline packages/antd/src/hooks/form/useDrawerForm/useDrawerForm.ts:114]: IF I'm not missing something, can't we refactor this like following suggestion?

```suggestion
            const openStatus = Boolean(parsed?.params?.[syncWithLocationKey]?.open);
```

File: packages/antd/src/hooks/form/useDrawerForm/useDrawerForm.ts [hunk 2]
```diff
@@ -102,15 +109,15 @@ export const useDrawerForm = <
...
             if (typeof openStatus === "boolean") {
-                setOpen(openStatus);
+                show();
```
[the remark this answers is not in the document]
Comment (salihozdemir) [reply packages/antd/src/hooks/form/useDrawerForm/useDrawerForm.ts:116]: Nice catch, we should care value, I'm fixing it here 👍

File: packages/antd/src/hooks/form/useDrawerForm/useDrawerForm.ts (hunk 1 above)
Comment (salihozdemir) [reply packages/antd/src/hooks/form/useDrawerForm/useDrawerForm.ts:114]: We can, but the code is handled for both boolean and string values. So it's okay if we don't.

File: packages/antd/src/hooks/form/useDrawerForm/index.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        });
+
+        await waitFor(() => result.current.mutationResult.isSuccess);
+
+        expect(result.current.formProps.form?.getFieldsValue()).toStrictEqual(
+            {},
+        );
+    });
+
+    fit("autoResetForm is false, 'reset' should not be called when the form is submitted", async () => {
```
Comment (omeraplak) [inline packages/antd/src/hooks/form/useDrawerForm/index.spec.tsx:167]:
```suggestion
    it("autoResetForm is false, 'reset' should not be called when the form is submitted", async () => {
```

**Review by omeraplak — APPROVED** [review]

**Pull request merged**